packages/scene-composer/src/components/panels/scene-settings/MatterportTagSync.tsx

```diff
@@ -40,10 +45,15 @@ export const MatterportTagSync: React.FC = () => {
     const layerName = MATTERPORT_TAG_LAYER_PREFIX + matterportModelId;
     let layerId = sceneLayerIds?.at(0);
     let rootId = sceneRootId;
+    const sceneMetadataModule = getGlobalSettings().twinMakerSceneMetadataModule;
```

Comment:
for most cases when `twinMakerSceneMetadataModule` is used to call APIs, it is triggered by a user action (like click the sync button). so we really just need to get the current version of `twinMakerSceneMetadataModule` and call API, no need to keep track of its change.

for `useDynamicScene` hook, it currently only check for the feature flag, but for later releases when we figured out how we want to determine if the dynamic scene feature is enabled for the scene/node, I want to add those logic to it as well. so I created the separate hook instead of `useFeature` directly.